mantidproject/mantid
# Use file name rather than nxs tag in SANS

**Description of work.**
    Uses the file name's run number rather than trying to pull it out of the
    nxs tags. This caused problems when data was transplanted from one file
    into the other, as the original name would be present despite the
    recipient taking the new name.
    
Additionally, the mock SANS instrument class was moved out of production code, since it caused warnings and should never have been included


**To test:**
- Obtain the ISIS training data and go to the LOQDemo folder
- Rename one of the files to something like LOQ71000
- Open the SANS GUI
- Ensure your default save dir is set
- Open the user file `maskfile.txt` from the LOQ data
- Enter run number 71000 into the first row of the table
- At the bottom change save options to both
- Hit process all, check that the output ws name has 710000 in it
- Go to the save dir and check the output file names also use 710000 in them

Fixes #27298

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
